sim/midas/src/main/scala/midas/widgets/Widget.scala

```diff
@@ -187,18 +169,96 @@ abstract class WidgetImp(wrapper: Widget) extends LazyModuleImp(wrapper) {
     crFile
   }
 
+
   /** Emits a header snippet for this widget.
     * @param base
     *    The base address of the MMIO region allocated to the widget.
     * @param memoryRegions
     *    A mapping of names to allocated FPGA-DRAM regions. This is one mechanism
     *    for establishing side-channels between two otherwise unconnected bridges or widgets.
     */
-  def genHeader(base: BigInt, memoryRegions: Map[String, BigInt], sb: StringBuilder): Unit = {
-    wrapper.headerComment(sb)
-    crRegistry.genHeader(wrapper.getWName.toUpperCase, base, sb)
-    crRegistry.genArrayHeader(wrapper.getWName.toUpperCase, base, sb)
-    wrapper.getHeaderFragments(base).foreach { sb.append(_) }
+  def genHeader(base: BigInt, memoryRegions: Map[String, BigInt], sb: StringBuilder): Unit
+
+  /** Emits a call to the construction into the generated header.
+    * @param base
+    *    The base address of the MMIO region allocated to the widget.
+    * @param sb
+    *    The string builder to append to.
+    * @param name
+    *    Name of the bridge driver class.
+    * @param headerName
+    *    Name of the header to get the driver from.
+    * @param args
+    *    List of C++ literals to pass as arguments.
+    * @param guard
+    *    Name of the header guard, used to order constructor calls.
+    * @param hasStreamEngine
+    *    Flag indicating that a stream engine reference should be provided.
+    * @param hasLoadMem
+    *    Flag indicating that a loadmem widget reference should be provided.
+    * @param hasAddrMap
+    *    Flag indicating that an address map should be provided.
+    */
+  def genConstructor(
+      base: BigInt,
+      sb: StringBuilder,
+      name: String,
+      headerName: String,
+      args: Seq[CPPLiteral] = Seq(),
+      guard: String = "GET_BRIDGE_CONSTRUCTOR",
+      hasStreamEngine: Boolean = false,
+      hasLoadMem: Boolean = false,
+      hasAddrMap: Boolean = false,
+  ): Unit = {
+    val mmioName = wrapper.getWName.toUpperCase.split("_")(0)
+    val regs = crRegistry.getSubstructRegs
+
+    sb.append("#ifdef GET_INCLUDES\n")
+    sb.append(s"#include ${'\"'}bridges/${headerName}.h${'\"'}\n")
```

Comment:
Use triple quotes here, or see below: 
```suggestion
    sb.append(s"""#include "bridges/${headerName}.h""" + "\n")
```